py-polars/polars/dataframe/frame.py

```diff
@@ -7064,6 +7065,10 @@ def join(
             .. note::
                 Joining on any other expressions than `col`
                 will turn off coalescing.
+        maintain_order
+            Whether to preserve the row order of the left DataFrame.
```

Comment:
Then the description must also be updated as we now allow any side to be order maintaining.